filebeat/harvester/reader/multiline.go

```diff
@@ -156,37 +158,37 @@ func (mlr *Multiline) readNext() (Message, error) {
 			}
 
 			// handle error without any bytes returned from reader
-			if message.Bytes == 0 {
-				// no lines buffered -> return error
-				if mlr.numLines == 0 {
-					return Message{}, err
-				}
-
-				// lines buffered, return multiline and error on next read
-				m := mlr.finalize()
-				mlr.err = err
-				mlr.setState((*Multiline).readFailed)
-				return m, nil
-			}
+			//if message.Bytes == 0 {
+			//	// no lines buffered -> return error
+			//	if mlr.numLines == 0 {
+			//		return Message{}, err
+			//	}
+			//
+			//	// lines buffered, return multiline and error on next read
+			//	m := mlr.finalize()
+			//	mlr.err = err
+			//	mlr.setState((*Multiline).readFailed)
+			//	return m, nil
+			//}
 
 			// handle error with some content being returned by reader and
 			// line matching multiline criteria or no multiline started yet
-			if mlr.readBytes == 0 || mlr.pred(mlr.last, message.Content) {
-				mlr.addLine(message)
-
-				// return multiline and error on next read
-				m := mlr.finalize()
-				mlr.err = err
-				mlr.setState((*Multiline).readFailed)
-				return m, nil
-			}
-
-			// no match, return current multline and retry with current line on next
+			//if mlr.readBytes == 0 || mlr.pred(mlr.last, message.Content) {
+			//	mlr.addLine(message)
+			//
+			//	// return multiline and error on next read
+			//	m := mlr.finalize()
+			//	mlr.err = err
+			//	mlr.setState((*Multiline).readFailed)
+			//	return m, nil
+			//}
+
+			// no match, return current multiline and retry with current line on next
 			// call to readNext awaiting the error being reproduced (or resolved)
 			// in next call to Next
 			m := mlr.finalize()
 			mlr.resetBuffer(message)
-			return m, nil
+			return m, err
```

Comment:
this line make clear, a reader can return a valid message plus error at the same time? Here potentially partial event will be pushed? Is it valid to reuse multiline reader after error?